sdk/core/azure-core-experimental/src/main/java/com/azure/core/experimental/jsonpatch/JsonPatchDocument.java

```diff
@@ -0,0 +1,127 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.experimental.jsonpatch;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Represents a JSON Patch document.
+ */
+public class JsonPatchDocument {
```

Comment:
Can we add some tests and samples to show how this can be used?

Reply:
Will add tests and samples in a subsequent commit.